keras/layers/containers.py

```diff
@@ -179,19 +195,10 @@ def updates(self):
                 updates += l.get_params()[3]
         return updates
 
-    def set_previous(self, layer, connection_map={}):
-        if self.nb_input != layer.nb_output:
-            raise Exception('Cannot connect layers: input count does not match output count.')
-        if self.nb_input == 1:
-            self.inputs[self.input_order[0]].set_previous(layer)
-        else:
-            if not connection_map:
-                raise Exception('Cannot attach multi-input layer: no connection_map provided.')
-            for k, v in connection_map.items():
-                if k in self.inputs and v in layer.outputs:
-                    self.inputs[k].set_previous(layer.outputs[v])
-                else:
-                    raise Exception('Invalid connection map.')
+    def set_previous(self, layer):
```

Comment:
This removes the `connection_map` wasn't it being used somewhere else already? Can you keep it for backward compatibility?